x-pack/plugins/index_lifecycle_management/__jest__/policy_table.test.tsx

```diff
@@ -198,6 +201,24 @@ describe('policy table', () => {
     });
   });
 
+  test('shows deprecated policies with Deprecated badges', () => {
+    const rendered = mountWithIntl(component);
+
+    // Initially the switch is off so we should not see any deprecated policies
+    let deprecatedPolicies = findTestSubject(rendered, 'deprecatedPolicyBadge');
+    expect(deprecatedPolicies.length).toBe(0);
+
+    // Enable filtering by deprecated policies
+    const searchInput = rendered.find('.euiFieldSearch').first();
+    (searchInput.instance() as unknown as HTMLInputElement).value = 'is:policy.deprecated';
+    searchInput.simulate('keyup', { key: 'Enter', keyCode: 13, which: 13 });
+    rendered.update();
+
+    // Now we should see all deprecated policies
+    deprecatedPolicies = findTestSubject(rendered, 'deprecatedPolicyBadge');
+    expect(deprecatedPolicies.length).toBe(10);
```

Comment:
nit: I'm wondering if it would be a good idea to expect the length to be greater than 0 rather than exactly 10 (so that we future proof the test in case some managed policies are added/removed in the test environment)?

Also nit, could we test that actually clicking the "Deprecated" button shows the deprecated policies?

Reply:
The table filters dont have a property for adding a data-test-subj, so the only way I could find how to test it was by inputting a filter in the search bar!